Sources/BraveWallet/Panels/Add Suggested Network/SuggestedNetworkView.swift

```diff
@@ -342,7 +342,13 @@ struct SuggestedNetworkView: View {
       customNetworkError = .failed(errorMessage: error)
       return
     }
-    onDismiss()
+    if let chain {
+      networkStore.customNetworkNativeAssetMigration(chain) {
+        onDismiss()
+      }
+    } else {
+      onDismiss()
+    }
```

Comment:
You could use the `onAddEthereumChainRequestCompleted` for this piece so it's not handled in this view, but noting that it would only cover `wallet_addEthereumChain` DApp requests and not when added from our Web3 settings.

Reply:
updated in [3ae6668](https://github.com/brave/brave-ios/pull/7625/commits/3ae66680a0580698e1bdd31a74d0ce0faf705cb9)